the-collab-lab/tcl-76-smart-shopping-list
# Ar mp mark item purchased

## Description

The app now keeps track of how many times a user purchases an item:
- We imported MUI to feature clean, simple checkboxes
- When the user checks the checkbox next to an item, Firebase counts it as a purchase, and the totalPurchase field increments by 1
- When the user checks the checkbox next to an item, Firebase updates the dateLastPurchased field to the time the checkbox was checked.
- After the user checks an item off (checks the checkbox), the box will stay checked for 24 hours, then uncheck itself after 24 hours. Firebase uses the dateLastPurchased timestamp to calculate 24 hours after. 

## Related Issue

closes 9

## Acceptance Criteria


## Type of Changes

`feature` `enhancement`

## Updates

### Before
- No checkboxes
<img width="768" alt="Screen Shot 2024-09-06 at 7 24 37 PM" src="https://github.com/user-attachments/assets/ee8c05bd-8045-4ab7-99c3-d45e2ab58ad8">



### After
- User can use checkboxes to mark when they've purchased that item.
- You can see before and after **avocado** is selected: dateLastPurchased updates, and totalPurchases increments up by 1. 
<img width="1277" alt="Screen Shot 2024-09-06 at 7 18 19 PM" src="https://github.com/user-attachments/assets/9f9e6549-6034-42ba-8a84-4aabff3b2661">

<img width="1162" alt="Screen Shot 2024-09-06 at 7 22 00 PM" src="https://github.com/user-attachments/assets/db047611-a84d-477c-b9ba-6881533528cb">

<img width="1377" alt="Screen Shot 2024-09-06 at 7 18 39 PM" src="https://github.com/user-attachments/assets/ef7bf5b1-9c88-4780-ac88-3d088b6a36b3">

## Testing Steps / QA Criteria

- Pull down from branch
- `npm i` to ensure MUI is installed properly
- open Firestore Database in a separate tab
- Navigate to your unique user code and a list of your choice, and the items inside that list
- Select an item (in my example, I selected **avocado**)
- check the checkbox next to that item
- See `dateLastPurchased` field update to the time you checked the checkbox (or, the current time)
- See the `totalPurchases` increase by 1.

To watch the checkbox uncheck itself in real time: 
- Have browser and Firestore tabs open next to each other
- Manually edit the `dateLastPurchased` field by clicking the pencil next to the field: Edit the field to the day + one minute before the current time (You can mess around with this!)
